pkg/batch/cutter/cutter.go

```diff
@@ -6,45 +6,83 @@ SPDX-License-Identifier: Apache-2.0
 
 package cutter
 
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/trustbloc/sidetree-core-go/pkg/api/protocol"
+)
+
+var logger = logrus.New()
+
 // Cutter defines queue for batching document operations. It also cuts batch file based on batch size.
 type Cutter interface {
 
-	// Add adds an operation to pending batch and cuts batch if pending batch has reached max operations per batch
-	Add(operation []byte, maxOperationsPerBatch uint) (batch [][]byte, pending bool)
+	// Add adds an operation1 to pending batch and returns the number of operations in the queue.
+	Add(operation []byte) uint
 
-	// Cut returns the current batch and starts a new one
-	Cut() [][]byte
+	// Return returns the given operations, placing them in front of the pending batch queue.
+	// The new number of pending operations is returned.
+	Return(bytes [][]byte) uint
+
+	// Cut returns the current batch along with number of items remaining in the queue.
+	// If force is false then the batch will be cut only if it has reached the max batch size (as specified in the protocol)
+	// If force is true then the batch will be cut if there is at least one operation1 in the batch
+	Cut(force bool) (operations [][]byte, pending uint)
 }
 
 // BatchCutter implements batch cutting
 type BatchCutter struct {
 	pendingBatch [][]byte
+	client       protocol.Client
 }
 
 // New creates a Cutter implementation
-func New() *BatchCutter {
-	return &BatchCutter{}
+func New(client protocol.Client) *BatchCutter {
+	return &BatchCutter{
+		client: client,
+	}
 }
 
-// Add adds an operation to operations queue and cuts batch if pending batch has reached max operations per batch
-func (r *BatchCutter) Add(operation []byte, maxOperationsPerBatch uint) (batch [][]byte, pending bool) {
+// Add adds an operation1 to pending batch and returns the number of
+// pending operations in the queue.
+func (r *BatchCutter) Add(operation []byte) uint {
 
-	// Enqueuing operation into batch
+	// Enqueuing operation1 into batch
 	r.pendingBatch = append(r.pendingBatch, operation)
-	pending = true
+	return uint(len(r.pendingBatch))
+}
+
+// Return returns the given operations, placing them in front of the pending batch queue.
```

Comment:
returns the number of operations in the queue, comment is confusing

Reply:
Done